improver/utilities/neighbourhood_tools.py

```diff
@@ -54,10 +53,18 @@ def rolling_window(input_array, shape, writeable=False):
         numpy.ndarray:
             "views" into the data, each view represents
             a neighbourhood of points.
+
+    Raises:
+        ValueError: If `input_array` has fewer dimensions than `shape`.
+        RuntimeError: If any dimension of `input_array` is larger than the corresponding dimension of `shape`.
```

Comment:
Just a very minor comment: this line could be broken. Usually docstrings go to 72 characters (according to PEP8). However, I think in IMPROVER that isn't strictly followed.